rust/src/transformations/make_stable_expr/namespace_str/mod.rs

```diff
@@ -0,0 +1,52 @@
+use polars::prelude::{FunctionExpr, StringFunction};
+use polars_plan::dsl::Expr;
+
+use crate::core::{MetricSpace, Transformation};
+use crate::domains::{ExprDomain, OuterMetric, WildExprDomain};
+use crate::error::*;
+use crate::polars::get_disabled_features_message;
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+#[cfg(feature = "contrib")]
+mod expr_strptime;
+
+/// Make a Transformation that returns an expression under the `str` namespace.
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The str expression
+pub fn make_namespace_str<M: OuterMetric>(
+    input_domain: WildExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<WildExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (WildExprDomain, M): MetricSpace,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Function {
+        function: FunctionExpr::StringExpr(str_function),
+        ..
+    } = &expr
+    else {
+        return fallible!(MakeTransformation, "expected cast expression");
+    };
+
+    match str_function {
+        #[cfg(feature = "contrib")]
+        StringFunction::Strptime(_, _) => expr_strptime::make_expr_strptime(input_domain, input_metric, expr),
+
+        expr => fallible!(
+            MakeTransformation,
+            "Expr is not recognized at this time: {:?}. {}If you would like to see this supported, please file an issue.",
+            expr,
+            get_disabled_features_message()
+        )
```

Comment:
This error seems to be conflating two separate things. Maybe they just don't have `contrib` enabled? Or maybe even with `contrib`, it doesn't match?

Maybe have two separate `fallible!`s, one guarded with `#[cfg(feature = "contrib")]`, and one with the negation?

Or if it's better to keep it simple, one error message that makes it clear that there are two different possible root causes?

Reply:
This is what the `get_disabled_features_message()` call does. If there are disabled features it interpolates that explanation into the error message.